pandas/io/formats/format.py

```diff
@@ -696,9 +696,8 @@ def _truncate_vertically(self) -> None:
         assert self.max_rows_fitted is not None
         row_num = self.max_rows_fitted // 2
         if row_num >= 1:
-            head = self.tr_frame.iloc[:row_num, :]
-            tail = self.tr_frame.iloc[-row_num:, :]
-            self.tr_frame = concat((head, tail))
+            dropped_indices = self.tr_frame.index[row_num:-row_num]
+            self.tr_frame = self.tr_frame.drop(dropped_indices)
```

Comment:
This doesn't work when there are duplicate indices:

```
size = 100
row_num = 10
tr_frame = pd.DataFrame(range(size), index=[1] * size)
dropped_indices = tr_frame.index[row_num:-row_num]
print(tr_frame.drop(dropped_indices))
# Empty DataFrame
# Columns: [0]
# Index: []
```

but I like the idea! What about something like:

```
rows = np.arange(len(self.tr_frame)
self.tr_frame = self.tr_frame.loc[(rows < row_num) | (rows >= len(tr_frame) - row_num)]
```

Reply:
yes i thought about something like this. But is `arange` memory efficient? or will this potentially double the memory used whenever we print? would that be a problem?

Reply:
also shouldn't it be `iloc` instead of `loc`?

Reply:
No it's not a concern. If generating an arange the length of your DataFrame gives you problems, you will very likely run into problems doing just about anything with pandas :smile:.

Reply:
Both loc and iloc allow a Boolean array.